# Description:
#   name: diag name
#   sf_at: diag level where it becomes Safe Fault
#   lf_at: diag level where it becomes Latent Fault
#   spf_at: diag level where it becomes Single Point Fault
#
# Note:
# empty-value for sf_at, lf_at and spf_at is "none"
# default values are:
#   sf_at: "none"
#   lf_at: "warn"
#   spf_at: "error"
---
/**:
  ros__parameters:
    required_modules:
      autonomous_driving:
        names: [
            "/autoware/control/autonomous_driving/alive_monitoring",
            "/autoware/control/autonomous_driving/lane_departure",
            # '/autoware/control/autonomous_driving/trajectory_deviation', # Unstable
            "/autoware/control/command_gate/alive_monitoring",
            "/autoware/localization/alive_monitoring",
            "/autoware/localization/matching_score",
            "/autoware/localization/localization_accuracy",
            "/autoware/map/alive_monitoring",
            "/autoware/map/map_version",
            "/autoware/perception/alive_monitoring",
            "/autoware/planning/alive_monitoring",
            "/autoware/sensing/alive_monitoring",
            "/autoware/sensing/camera",
            "/autoware/sensing/gnss",
            "/autoware/sensing/imu",
            "/autoware/sensing/lidar",
            "/autoware/system/alive_monitoring",
            "/autoware/system/emergency_stop_operation",
            "/autoware/system/logging",
            # '/autoware/system/resource_monitoring/cpu_temperature', # Not working in containers
            # '/autoware/system/resource_monitoring/cpu_usage', # No need to watch in simulation
            # '/autoware/system/resource_monitoring/cpu_thermal_throttling', # Require setup
            # '/autoware/system/resource_monitoring/hdd_temperature', # Require setup
            # '/autoware/system/resource_monitoring/hdd_usage', # Not working in containers
            # '/autoware/system/resource_monitoring/memory_usage', # No need to watch in simulation
            # '/autoware/system/resource_monitoring/network_usage', # No need to watch in simulation
            # '/autoware/system/resource_monitoring/ntp_offset', # No need to watch in simulation
            # '/autoware/system/resource_monitoring/gpu_temperature', # Require GPU
            # '/autoware/system/resource_monitoring/gpu_usage', # Require GPU
            # '/autoware/system/resource_monitoring/gpu_memory_usage', # Require GPU
            # '/autoware/system/resource_monitoring/gpu_thermal_throttling', # Require GPU
            "/autoware/vehicle/alive_monitoring",
            "/autoware/vehicle/vehicle_errors",
          ]
        diag_level:
          /autoware/localization/matching_score: { sf_at: "warn", lf_at: "error", spf_at: "none" }
          /autoware/sensing/gnss: { sf_at: "warn", lf_at: "error", spf_at: "none" }
          /autoware/system/logging: { sf_at: "warn", lf_at: "none", spf_at: "none" }
          # /autoware/system/resource_monitoring/cpu_temperature: { sf_at: "warn", lf_at: "error", spf_at: "none" } # Not working in VMs
          # /autoware/system/resource_monitoring/cpu_usage: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/cpu_thermal_throttling: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/hdd_temperature: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/hdd_usage: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/memory_usage: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/network_usage: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/ntp_offset: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/gpu_temperature: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/gpu_usage: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/gpu_memory_usage: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation
          # /autoware/system/resource_monitoring/gpu_thermal_throttling: { sf_at: "warn", lf_at: "error", spf_at: "none" }  # No need to watch in simulation

      remote_control:
        names:
          [
            "/autoware/control/command_gate/alive_monitoring",
            "/autoware/control/remote_control/alive_monitoring",
            "/autoware/system/alive_monitoring",
            "/autoware/system/emergency_stop_operation",
            "/autoware/vehicle/alive_monitoring",
            "/autoware/vehicle/vehicle_errors",
          ]
